mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2117,3 +2215,49 @@ where
     unspents.dedup_by(|one, another| one.outpoint == another.outpoint);
     Ok((unspents, recently_spent))
 }
+
+async fn merge_utxo_loop<T, W>(weak: W, merge_at: usize, check_every: f64, max_merge_at_once: usize)
+where
+    W: WeakUpgradableCoin<Target = T>,
```

Comment:
The WeakUpgradableCoin/ArcDowngradableCoin concept looks quite excessive to me. The previous merge_utxo_loop implementation problem was using the UtxoStandardCoin, but the loop could be started for another coin type. It, in turn, could lead to a call of the wrong list_unspent_ordered version. 
Can you simply require `T: From<UtxoArc> + AsRef<UtxoCoinFields> + UtxoCommonOps` and then do 
```
let coin = match weak.upgrade() {
    Some(a) => T::from(a),
    None => break,
};
```

I guess you can remove the `WeakUpgradableCoin/ArcDowngradableCoin` then.

Reply:
Done, thanks :)